build.rs

```diff
@@ -161,28 +161,19 @@ fn compile_simd_c() {
     let mut build = cc::Build::new();
     build.opt_level(3);
 
-    // `-march=native` is not supported on Apple M1
-    #[cfg(not(all(target_os = "macos", target_arch = "aarch64")))]
-    {
-        fn guess_target_cpu() -> String {
-            // Copied and adapted from https://github.com/alexcrichton/proc-macro2/blob/4173a21dc497c67326095e438ff989cc63cd9279/build.rs#L115
-            // Licensed under Apache-2.0 + MIT (compatible because we're MIT)
-            let rustflags = env::var_os("RUSTFLAGS").or(env::var_os("CARGO_ENCODED_RUSTFLAGS"));
-            if let Some(rustflags) = rustflags {
-                for mut flag in rustflags.to_string_lossy().split(&[' ', '\u{1f}'][..]) {
-                    if flag.starts_with("-C") {
-                        flag = &flag["-C".len()..];
-                    }
-                    if flag.starts_with("target-cpu=") {
-                        return flag["target-cpu=".len()..].to_owned();
-                    }
-                }
-            }
-
-            "native".to_string()
+    match env::var("RSE_ARCH") {
```

Comment:
How about including a `Cargo.toml` feature as well, perhaps called `target-cpu-native`.  This'll allow the user to retain the "native" cpu target without the need for an environment variable (`RSE_ARCH`).  Env vars can be clumsy to integrate into downstream build processes, especially those that aim for a simple `cargo build` workflow

Reply:
Simple `cargo build` already works and retains over 90% of performance on x86-64. Having `native` feels fragile to have in general and shouldn't be recommended because binaries become non-portable.

I can add a feature for that if you insist, but I don't see how `RSE_ARCH=native cargo build` is much harder than `cargo build` or those who care about it.

Reply:
In a high performance use case where binary portability is not a concern, leaving 10% on the table kinda sucks.

A new env var is definitely much more error prone than a cargo toml feature.

Reply:
> In a high performance use case where binary portability is not a concern, leaving 10% on the table kinda sucks.

It's not like this performance boost will be lost. You can always specify `RSE_ARCH=native` to those 10% back.

> A new env var is definitely much more error prone than a cargo toml feature.

This doesn't feel like a feature though, as it doesn't change code itself, but rather compilation flags.

Reply:
> It's not like this performance boost will be lost. You can always specify RSE_ARCH=native to those 10% back.

Right. My concern is the env var setting itself as that is an extra step that users will forget from time to time or necessitates some kind of project-specific `cargo` wrapper to always set the var.  This is I was exploring using a Cargo feature for it since the downstream user could just configure this once in their Cargo.toml and forget about it, regardless of whether this is technically is a "feature" or not :)

Reply:
Usually performance-sensitive builds are automated, forgetting environment there is possible of course, but less likely.

Reply:
Ok this PR is strictly better than what's currently in master so let's get it landed. There can always be a follow-up if the env var is too unwieldy.

I'd love to see `RSE_ARCH` turn into something much more verbose though. "RSE" is not really a well-known acronym so it's going to confuse some for sure.  Perhaps `RUST_REED_SOLOMON_ERASURE_ARCH`?

Reply:
Done, please take a look one more time